src/db-service/shopping_cart.js

```diff
@@ -77,8 +77,8 @@ exports.getCart = async (cartId, format = true) => {
 };
 
 // I was unsure how to handle item that had no inventory so for now I am leaving them in the cart :)
-exports.checkoutCart = async products => {
```

Comment:
Consider finding a better name than checkoutCart.

Reply:
I’m a bit worried about the overhead that this comes with… it means 1 extra pointer field for each cleanup hook instance.

If the goal is to extend this to more classes, how moving DeleteMe from BaseObject to MemoryRetainer, and allow other subclasses of MemoryRetainer to use that as well?

Alternatively, it might make sense to put BaseObjects/MemoryRetainers into a separate data structure that doesn’t need the fn_ field at all, or use an intrusive linked list (like HandleWrapQueue)?